lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -88,7 +91,13 @@ def define_virtual_aggregate_method(name, relation, method_name, column)
           if attribute_present?(name)
             self[name]
           elsif (rel = send(relation)).loaded?
-            rel.blank? ? nil : rel.map { |t| t.send(column) }.compact.send(method_name)
+            values = rel.map { |t| t.send(column) }.compact
+            return nil if values.blank?
```

Comment:
```suggestion
            return nil if values.empty?
```

same thing but reads better to me. your choice

Reply:
also think this one's an old diff, this is the current:
```ruby
if has_attribute?(name)
            self[name] || 0
          elsif (rel = send(relation)).loaded?
            values = rel.map { |t| t.send(column) }.compact
            if block_given?
              yield values
            else
              rel.blank? ? nil : values.send(ruby_method_name)
            end
          else
            rel.try(arel_method_name, column) || 0
          end
```